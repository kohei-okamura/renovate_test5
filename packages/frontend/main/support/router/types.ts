/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Route } from 'vue-router'

export type RouteParams = Route['params']

export type RouteQuery = Route['query']
