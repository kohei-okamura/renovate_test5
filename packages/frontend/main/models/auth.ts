/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Permission } from '@zinger/enums/lib/permission'
import { Staff } from '~/models/staff'

/**
 * 認証・認可情報.
 */
export type Auth = Readonly<{
  /** システム管理者フラグ */
  isSystemAdmin: boolean

  /** 権限 */
  permissions: Permission[]

  /** スタッフ */
  staff: Staff
}>
