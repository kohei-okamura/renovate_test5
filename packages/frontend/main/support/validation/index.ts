/* eslint-disable camelcase */
/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { configure, extend, localize, ValidationObserver, ValidationProvider } from 'vee-validate'
import {
  alpha_num as alphaNum,
  between,
  digits,
  email,
  integer,
  length,
  max,
  max_value as maxValue,
  min,
  min_value as minValue,
  numeric,
  required
} from 'vee-validate/dist/rules'
import { VueConstructor } from 'vue'
import { asciiAlphaNum } from '~/support/validation/rules/ascii-alpha-num'
import { custom } from '~/support/validation/rules/custom'
import { fax } from '~/support/validation/rules/fax'
import { hiragana } from '~/support/validation/rules/hiragana'
import { katakana } from '~/support/validation/rules/katakana'
import { minDate } from '~/support/validation/rules/min-date'
import { nonItemsZero } from '~/support/validation/rules/non-items-zero'
import { postcode } from '~/support/validation/rules/postcode'
import { tel } from '~/support/validation/rules/tel'
import { validTimeDuration } from '~/support/validation/rules/valid-time-durations'
import { zenginDataRecordChar } from '~/support/validation/rules/zengin-data-record-char'

function setupComponents (vue: VueConstructor): void {
  vue.component('ValidationObserver', ValidationObserver)
  vue.component('ValidationProvider', ValidationProvider)
}

function setupRules (): void {
  extend('alphaNum', alphaNum)
  extend('asciiAlphaNum', asciiAlphaNum)
  extend('between', between)
  extend('custom', custom)
  extend('digits', digits)
  extend('email', email)
  extend('fax', fax)
  extend('hiragana', hiragana)
  extend('integer', integer)
  extend('katakana', katakana)
  extend('length', length)
  extend('max', max)
  extend('maxValue', maxValue)
  extend('min', min)
  extend('minDate', minDate)
  extend('minValue', minValue)
  extend('nonItemsZero', nonItemsZero)
  extend('numeric', numeric)
  extend('postcode', postcode)
  extend('required', required)
  extend('tel', tel)
  extend('validTimeDuration', validTimeDuration)
  extend('zenginDataRecordChar', zenginDataRecordChar)
}

function setupMessages (): void {
  localize('ja', {
    messages: {
      alphaNum: '半角英数字で入力してください',
      between: (_, params = {}) => `${params.min}以上、${params.max}以下の半角数字で入力してください。`,
      digits: (_, params = {}) => `${params.length}桁の半角数字で入力してください。`,
      email: '有効なメールアドレスを入力してください。',
      integer: '半角数字のみで入力してください。',
      length: (_, params = {}) => `${params.length}文字で入力してください。`,
      max: (_, params = {}) => `${params.length}文字以内で入力してください。`,
      maxValue: (_, params = {}) => `${params.max}以下を入力してください。`,
      min: (_, params = {}) => `${params.length}文字以上で入力してください。`,
      minValue: (_, params = {}) => `${params.min}以上を入力してください。`,
      numeric: '半角数字のみで入力してください。',
      required: '入力してください。'
    }
  })
}

/**
 * VeeValidate 関連のセットアップ処理.
 */
export function setupVeeValidate (vue: VueConstructor): void {
  configure({
    mode: 'eager'
  })
  setupComponents(vue)
  setupRules()
  setupMessages()
}
