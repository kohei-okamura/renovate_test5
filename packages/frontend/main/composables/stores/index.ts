/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { createSessionStore } from '~/composables/stores/create-session-store'
import { NuxtContext } from '~/models/nuxt'

type CreateGlobalStoreParams = Pick<NuxtContext, '$api'>

export const createGlobalStore = (params: CreateGlobalStoreParams) => ({
  session: createSessionStore(params)
})

export type GlobalStore = ReturnType<typeof createGlobalStore>
