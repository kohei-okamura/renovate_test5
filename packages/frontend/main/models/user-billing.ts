/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { UserBillingResult } from '@zinger/enums/lib/user-billing-result'
import { WithdrawalResultCode } from '@zinger/enums/lib/withdrawal-result-code'
import { DateLike } from '~/models/date'
import { OfficeId } from '~/models/office'
import { UserId } from '~/models/user'
import { UserBillingDwsItem } from '~/models/user-billing-dws-item'
import { UserBillingLtcsItem } from '~/models/user-billing-ltcs-item'
import { UserBillingOffice } from '~/models/user-billing-office'
import { UserBillingOtherItem } from '~/models/user-billing-other-item'
import { UserBillingUser } from '~/models/user-billing-user'

/**
 * 利用者請求 ID.
 */
export type UserBillingId = number

/**
 * 利用者請求.
 */
export type UserBilling = Readonly<{
  /** 利用者請求 ID */
  id: UserBillingId

  /** 利用者 ID */
  userId: UserId

  /** 事業所 ID */
  officeId: OfficeId

  /** 利用者 */
  user: UserBillingUser

  /** 事業所 */
  office: UserBillingOffice

  /** 障害福祉サービス明細 */
  dwsItem: UserBillingDwsItem | undefined

  /** 介護保険サービス明細 */
  ltcsItem: UserBillingLtcsItem | undefined

  /** その他サービス明細 */
  otherItems: UserBillingOtherItem[]

  /** 請求結果 */
  result: UserBillingResult

  /** 合計金額 */
  totalAmount: number

  /** 繰越金額 */
  carriedOverAmount: number

  /** 振替結果コード */
  withdrawalResultCode: WithdrawalResultCode | undefined

  /** サービス提供年月 */
  providedIn: DateLike

  /** 発行日 */
  issuedOn: DateLike | undefined

  /** 入金日時 */
  depositedAt: DateLike | undefined

  /** 処理日時 */
  transactedAt: DateLike | undefined

  /** 口座振替日 */
  deductedOn: DateLike | undefined

  /** お支払期限日 */
  dueDate: DateLike | undefined

  /** 登録日時 */
  createdAt: DateLike

  /** 更新日時 */
  updatedAt: DateLike
}>
