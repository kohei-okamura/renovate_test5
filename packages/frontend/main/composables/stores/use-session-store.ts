/*
 * Copyright Â© 2020 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { InjectionKey } from '@nuxtjs/composition-api'
import { SessionState, SessionStore } from '~/composables/stores/create-session-store'
import { usePlugins } from '~/composables/use-plugins'

export function useSessionStore () {
  const { $globalStore } = usePlugins()
  return $globalStore.session
}

export const sessionStateKey: InjectionKey<SessionState> = Symbol('sessionState')

export const sessionStoreKey: InjectionKey<SessionStore> = Symbol('sessionStore')
