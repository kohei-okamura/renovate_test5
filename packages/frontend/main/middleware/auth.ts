/*
 * Copyright © 2019 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { Context } from '@nuxt/types'
import { Permission } from '@zinger/enums/lib/permission'
import { useAuth } from '~/composables/use-auth'

/**
 * 認証・認可ミドルウェア.
 */

type AuthMiddleware = {
  (...permissions: Permission[]): (context: Context) => Promise<void>
}

export const auth: AuthMiddleware = (...permissions) => async ({ app, redirect, route }: Context) => {
  const { session } = app.$globalStore
  const { prepare, isAuthenticated, isAuthorized } = useAuth(session)
  const redirectPromise = (path: string) => new Promise<void>(() => redirect(path))
  await prepare()
  if (!isAuthenticated.value) {
    // 未認証の場合はトップページにリダイレクトする
    // 画面を表示させないため下記の Promise は resolve しないこと
    return redirectPromise(`/?path=${route.fullPath}`)
  } else if (!isAuthorized.value(permissions)) {
    // 認可に失敗（＝必要な権限がない）場合は NotFound とする
    // 画面を表示させないため下記の Promise は resolve しないこと
    return redirectPromise('/notfound')
  } else {
    // 認証・認可に成功した場合はなにもしない
    return Promise.resolve()
  }
}
