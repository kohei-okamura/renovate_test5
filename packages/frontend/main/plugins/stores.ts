/*
 * Copyright Â© 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { defineNuxtPlugin, InjectionKey, onGlobalSetup, provide } from '@nuxtjs/composition-api'
import {
  attendancesStateKey,
  attendancesStoreKey,
  useAttendancesStore
} from '~/composables/stores/use-attendances-store'
import {
  billingBulkUpdateStateKey,
  billingBulkUpdateStoreKey,
  useBillingBulkUpdateStore
} from '~/composables/stores/use-billing-bulk-update-store'
import {
  dwsBillingsStateKey,
  dwsBillingsStoreKey,
  useDwsBillingsStore
} from '~/composables/stores/use-dws-billings-store'
import {
  dwsProvisionReportsIndexStateKey,
  dwsProvisionReportsIndexStoreKey,
  useDwsProvisionReportsStore
} from '~/composables/stores/use-dws-provision-reports-store'
import {
  ltcsBillingsStateKey,
  ltcsBillingsStoreKey,
  useLtcsBillingsStore
} from '~/composables/stores/use-ltcs-billings-store'
import {
  ltcsProvisionReportsIndexStateKey,
  ltcsProvisionReportsIndexStoreKey,
  useLtcsProvisionReportsStore
} from '~/composables/stores/use-ltcs-provision-reports-store'
import {
  notificationStateKey,
  notificationStoreKey,
  useNotificationStore
} from '~/composables/stores/use-notification-store'
import { officesStateKey, officesStoreKey, useOfficesStore } from '~/composables/stores/use-offices-store'
import {
  ownExpenseProgramsStateKey,
  ownExpenseProgramsStoreKey,
  useOwnExpenseProgramsStore
} from '~/composables/stores/use-own-expense-programs-store'
import { rolesStateKey, rolesStoreKey, useRolesStore } from '~/composables/stores/use-roles-store'
import { sessionStateKey, sessionStoreKey, useSessionStore } from '~/composables/stores/use-session-store'
import {
  shiftImportsStateKey,
  shiftImportsStoreKey,
  useShiftImportsStore
} from '~/composables/stores/use-shift-imports-store'
import { shiftsStateKey, shiftsStoreKey, useShiftsStore } from '~/composables/stores/use-shifts-store'
import { staffsStateKey, staffsStoreKey, useStaffsStore } from '~/composables/stores/use-staffs-store'
import {
  userBillingsStateKey,
  userBillingsStoreKey,
  useUserBillingsStore
} from '~/composables/stores/use-user-billings-store'
import { usersStateKey, usersStoreKey, useUsersStore } from '~/composables/stores/use-users-store'
import { StateProvider } from '~/composables/stores/utils'

type ProvideStore = <T extends Record<string, unknown>, U extends StateProvider<T>> (
  store: U,
  storeKey: InjectionKey<U>,
  stateKey: InjectionKey<T>
) => void

const provideStore: ProvideStore = (store, storeKey, stateKey) => {
  provide(storeKey, store)
  provide(stateKey, store.state)
}

export default defineNuxtPlugin(() => onGlobalSetup(() => {
  provideStore(useUserBillingsStore(), userBillingsStoreKey, userBillingsStateKey)
  provideStore(useLtcsBillingsStore(), ltcsBillingsStoreKey, ltcsBillingsStateKey)
  provideStore(useDwsBillingsStore(), dwsBillingsStoreKey, dwsBillingsStateKey)
  provideStore(useAttendancesStore(), attendancesStoreKey, attendancesStateKey)
  provideStore(useShiftsStore(), shiftsStoreKey, shiftsStateKey)
  provideStore(useOwnExpenseProgramsStore(), ownExpenseProgramsStoreKey, ownExpenseProgramsStateKey)
  provideStore(useOfficesStore(), officesStoreKey, officesStateKey)
  provideStore(useStaffsStore(), staffsStoreKey, staffsStateKey)
  provideStore(useUsersStore(), usersStoreKey, usersStateKey)
  provideStore(useDwsProvisionReportsStore(), dwsProvisionReportsIndexStoreKey, dwsProvisionReportsIndexStateKey)
  provideStore(useLtcsProvisionReportsStore(), ltcsProvisionReportsIndexStoreKey, ltcsProvisionReportsIndexStateKey)
  provideStore(useNotificationStore(), notificationStoreKey, notificationStateKey)
  provideStore(useRolesStore(), rolesStoreKey, rolesStateKey)
  provideStore(useSessionStore(), sessionStoreKey, sessionStateKey)
  provideStore(useShiftImportsStore(), shiftImportsStoreKey, shiftImportsStateKey)
  provideStore(useBillingBulkUpdateStore(), billingBulkUpdateStoreKey, billingBulkUpdateStateKey)
}))
