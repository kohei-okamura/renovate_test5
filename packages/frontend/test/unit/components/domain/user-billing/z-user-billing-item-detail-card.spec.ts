/*
 * Copyright © 2021 EUSTYLE LABORATORY - ALL RIGHTS RESERVED.
 * UNAUTHORIZED COPYING OF THIS FILE, VIA ANY MEDIUM IS STRICTLY PROHIBITED PROPRIETARY AND CONFIDENTIAL.
 */
import { MountOptions, Wrapper } from '@vue/test-utils'
import { Permission } from '@zinger/enums/lib/permission'
import { UserBillingResult } from '@zinger/enums/lib/user-billing-result'
import deepmerge from 'deepmerge'
import flushPromises from 'flush-promises'
import Vue from 'vue'
import ZUserBillingItemDetailCard from '~/components/domain/user-billing/z-user-billing-item-detail-card.vue'
import { sessionStoreKey } from '~/composables/stores/use-session-store'
import { Auth } from '~/models/auth'
import { UserBilling } from '~/models/user-billing'
import { ValidationObserverInstance } from '~/support/validation/types'
import { createUserBillingStub } from '~~/stubs/create-user-billing-stub'
import { createAuthStub } from '~~/test/helpers/create-auth-stub'
import { getValidationObserver } from '~~/test/helpers/get-validation-observer'
import { provides } from '~~/test/helpers/provides'
import { setData } from '~~/test/helpers/set-data'
import { setupComponentTest } from '~~/test/helpers/setup-component-test'

describe('z-user-billing-item-card.vue', () => {
  const { mount } = setupComponentTest()
  const userBilling = createUserBillingStub()

  const defaultPropsData = {
    userBilling: {
      ...userBilling,
      carriedOverAmount: -50,
      result: UserBillingResult.pending,
      totalAmount: 200
    }
  }

  let wrapper: Wrapper<Vue & any>

  type MountComponentArguments = MountOptions<Vue> & {
    auth?: Partial<Auth>
    propsData?: { userBilling: DeepPartial<UserBilling> }
  }

  function mountComponent ({ auth, propsData, ...options }: MountComponentArguments = {}) {
    wrapper = mount(ZUserBillingItemDetailCard, {
      propsData: deepmerge(defaultPropsData, propsData ?? {}),
      ...options,
      mocks: {
        ...options?.mocks
      },
      ...provides([sessionStoreKey, createAuthStub(auth ?? { isSystemAdmin: true })])
    })
  }

  function unmountComponent () {
    wrapper.destroy()
  }

  it('should be rendered correctly', () => {
    mountComponent()
    expect(wrapper).toMatchSnapshot()
  })

  describe('permission', () => {
    it('should be rendered when the staff have permissions.updateUserBillings', () => {
      mountComponent({ auth: { permissions: [Permission.updateUserBillings] } })
      expect(wrapper.find('[data-carried-over-amount-button]')).toExist()
      unmountComponent()
    })

    it('should not be rendered when the result is neither pending nor none', () => {
      mountComponent({ propsData: { userBilling: { result: UserBillingResult.paid } } })
      expect(wrapper.find('[data-carried-over-amount-button]')).not.toExist()
      unmountComponent()
    })

    it('should be rendered when the result is pending', () => {
      mountComponent({ propsData: { userBilling: { result: UserBillingResult.pending } } })
      expect(wrapper.find('[data-carried-over-amount-button]')).toExist()
      unmountComponent()
    })

    it('should be rendered when the result is none', () => {
      mountComponent({ propsData: { userBilling: { result: UserBillingResult.none } } })
      expect(wrapper.find('[data-carried-over-amount-button]')).toExist()
      unmountComponent()
    })

    it('should be rendered when the staff does not have permissions.updateUserBillings', () => {
      mountComponent({ auth: { permissions: [] } })
      expect(wrapper.find('[data-carried-over-amount-button]')).not.toExist()
      unmountComponent()
    })
  })

  describe('validation', () => {
    let observer: ValidationObserverInstance

    async function validate (values: { changedCarriedOverAmount?: number | string } = {}): Promise<void> {
      await setData(wrapper, values)
      await observer.validate()
      jest.runOnlyPendingTimers()
    }

    beforeAll(() => {
      mountComponent()
      observer = getValidationObserver(wrapper)
    })

    afterAll(() => {
      unmountComponent()
    })

    afterEach(() => {
      observer.reset()
    })

    it('should not fail if carriedOverAmount is negative number', async () => {
      const messageWrapper = wrapper.find('[data-carried-over-amount-form] .v-messages')
      expect(messageWrapper.text()).toBe('')
      await validate({ changedCarriedOverAmount: -1 })
      expect(observer).toBePassed()
    })

    it('should not fail if carriedOverAmount is zero', async () => {
      const messageWrapper = wrapper.find('[data-carried-over-amount-form] .v-messages')
      expect(messageWrapper.text()).toBe('')
      await validate({ changedCarriedOverAmount: 0 })
      expect(observer).toBePassed()
    })

    it('should not fail if carriedOverAmount plus totalAmount is zero', async () => {
      const messageWrapper = wrapper.find('[data-carried-over-amount-form] .v-messages')
      expect(messageWrapper.text()).toBe('')
      // FYI totalAmount には既に carriedOverAmount が含まれているため指定可能な繰越金額の上限は
      // totalAmount(200) - 現在の carriedOverAmount(-50) になる
      await validate({ changedCarriedOverAmount: -250 })
      expect(observer).toBePassed()
    })

    it('should fail if carriedOverAmount is string', async () => {
      const messageWrapper = wrapper.find('[data-carried-over-amount-form] .v-messages')
      expect(messageWrapper.text()).toBe('')
      await validate({ changedCarriedOverAmount: 'fail' })
      expect(observer).not.toBePassed()
      expect(messageWrapper.text()).toBe('半角数字のみで入力してください。')
    })

    it('should fail if carriedOverAmount plus totalAmount is negative number', async () => {
      const messageWrapper = wrapper.find('[data-carried-over-amount-form] .v-messages')
      expect(messageWrapper.text()).toBe('')
      await validate({ changedCarriedOverAmount: -251 })
      expect(observer).not.toBePassed()
      expect(messageWrapper.text()).toBe('繰越金額と各明細の合計金額の合計が0円以上になるようにしてください。')
    })

    it('should fail if carriedOverAmount is not changed', async () => {
      const messageWrapper = wrapper.find('[data-carried-over-amount-form] .v-messages')
      expect(messageWrapper.text()).toBe('')
      await validate({ changedCarriedOverAmount: -50 })
      expect(observer).not.toBePassed()
      expect(messageWrapper.text()).toBe('繰越金額を変更してください。')
    })
  })

  describe('event', () => {
    beforeAll(() => {
      mountComponent()
    })

    afterAll(() => {
      unmountComponent()
    })

    describe('update carried over amount', () => {
      it('should show dialog', async () => {
        const dialog = wrapper.findComponent({ name: 'z-prompt-dialog' })
        expect(dialog.props().active).toBeFalse()

        await wrapper.vm.showChangeDialog()
        await wrapper.vm.$nextTick()

        // props.active が true になっていることを確認する
        expect(dialog.props().active).toBeTrue()
      })

      it('should emit click:update when positive clicked', async () => {
        const value = '-100'

        await setData(wrapper, { changedCarriedOverAmount: value })
        await wrapper.vm.showChangeDialog()
        await wrapper.vm.$nextTick()

        const dialog = wrapper.findComponent({ name: 'z-prompt-dialog' })

        await dialog.vm.$emit('click:positive', new MouseEvent('click'))
        await flushPromises()

        const emitted = wrapper.emitted('click:update')
        expect(emitted).toBeTruthy()
        expect(emitted!.length).toBe(1)
        expect(emitted![0][0]).toStrictEqual(parseInt(value))
        expect(dialog.props().active).toBeFalse()
      })

      it('should not emit click:update when negative clicked', async () => {
        await wrapper.vm.showChangeDialog()
        await wrapper.vm.$nextTick()

        const dialog = wrapper.findComponent({ name: 'z-prompt-dialog' })

        await dialog.vm.$emit('click:negative', new MouseEvent('click'))
        await flushPromises()

        const emitted = wrapper.emitted('click:update')
        // FIXME
        // 毎回 mount をするのはコストが高いので避けたいが、それだとケースごとに emitted をクリアできない
        // ひとまず length が増えていないことで呼ばれていないことを確認しているが emitted をクリアしたい
        expect(emitted!.length).toBe(1)
        expect(dialog.props().active).toBeFalse()
      })
    })
  })
})
