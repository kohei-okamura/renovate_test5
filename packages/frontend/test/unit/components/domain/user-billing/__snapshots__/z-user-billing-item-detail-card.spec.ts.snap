// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`z-user-billing-item-card.vue should be rendered correctly 1`] = `
<div>
  <div
    class="mt-2 z-data-card-stub"
  >
    <div
      class="z-data-card-stub-slot-actions"
    />
    <z-data-card-item-stub
      icon="M13.92 11H18V13H13V15H18V17H13V21H11V17H6V15H11V13H6V11H10.08L5 3H7.37L12 10.29L16.63 3H19L13.92 11Z"
      label="税抜金額(10%)"
      value="144,899円"
    />
    <z-data-card-item-stub
      label="消費税額(10%)"
      value="14,487円"
    />
    <z-data-card-item-stub
      label="税抜金額(8%)"
      value="0円"
    />
    <z-data-card-item-stub
      label="消費税額(8%)"
      value="0円"
    />
    <z-data-card-item-stub
      label="繰越金額"
      value="-50円"
    />
    <z-data-card-item-stub
      label="合計"
      value="200円"
    />
    <z-data-card-item-stub
      label="医療費控除対象額"
      value="288,145円"
    />
    <v-card-actions-stub
      tag="div"
    >
      <v-spacer-stub
        tag="div"
      />
      <a
        class="v-btn-stub"
        color="primary"
        data-carried-over-amount-button=""
        text=""
      >
        繰越金額を変更する
      </a>
    </v-card-actions-stub>
    <div
      class="z-data-card-stub-slot-header"
    />
  </div>
  <v-dialog-stub
    closedelay="0"
    contentclass=""
    data-carried-over-amount-dialog=""
    opendelay="0"
    origin="center center"
    persistent="true"
    retainfocus="true"
    transition="dialog-transition"
    width="550"
  >
    <v-card-stub
      loaderheight="4"
      tag="div"
    >
      <!---->
      <v-card-text-stub
        tag="div"
      >
        <p
          class="message"
        >
          繰越金額を入力して実行を押してください。
        </p>
        <div>
          <z-data-card-item-stub
            icon="M13.92 11H18V13H13V15H18V17H13V21H11V17H6V15H11V13H6V11H10.08L5 3H7.37L12 10.29L16.63 3H19L13.92 11Z"
            label="合計"
            value="200円"
          />
          <z-data-card-item-stub
            label="現在の繰越金額"
            value="-50円"
          />
          <z-form-card-item-set-stub
            icon=""
            label=""
          >
            <div
              class="root"
              data-carried-over-amount-form=""
              data-z-form-card-item=""
            >
              <div
                class="v-input z-text-field--numeric v-input--is-label-active v-input--is-dirty v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined textField"
              >
                <div
                  class="v-input__control"
                >
                  <div
                    class="v-input__slot"
                    style="background-color: rgb(241, 246, 250); border-color: #f1f6fa;"
                  >
                    <fieldset
                      aria-hidden="true"
                    >
                      <legend
                        style="width: 0px;"
                      >
                        <span
                          class="notranslate"
                        >
                          ​
                        </span>
                      </legend>
                    </fieldset>
                    <div
                      class="v-text-field__slot"
                    >
                      <label
                        class="v-label v-label--active theme--light"
                        style="left: 0px; position: absolute;"
                      >
                        変更後の繰越金額 *
                      </label>
                      <input
                        type="text"
                      />
                      <div
                        class="v-text-field__suffix"
                      >
                        円
                      </div>
                    </div>
                  </div>
                  <div
                    class="v-text-field__details"
                  >
                    <div
                      class="v-messages theme--light"
                    >
                      <transition-group-stub
                        class="v-messages__wrapper"
                        name="message-transition"
                        tag="div"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </z-form-card-item-set-stub>
        </div>
      </v-card-text-stub>
      <v-card-actions-stub
        tag="div"
      >
        <v-spacer-stub
          tag="div"
        />
        <a
          class="v-btn-stub"
          color="grey"
          data-button-negative=""
          text=""
        >
          <span>
            キャンセル
          </span>
        </a>
        <a
          class="v-btn-stub"
          color="primary"
          data-button-positive=""
          text=""
          type="submit"
        >
          <span>
            保存
          </span>
        </a>
      </v-card-actions-stub>
    </v-card-stub>
  </v-dialog-stub>
</div>
`;
